File: repro_upload.js
const fs = require('fs');
const path = require('path');
const { Blob } = require('buffer');

async function main() {
  const filePath = path.join(process.cwd(), 'tmp.png');
  if (!fs.existsSync(filePath)) {
    const b = Buffer.from('iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAACklEQVR4nGMAAIAAeIh6xwAAAABJRU5ErkJggg==', 'base64');
    fs.writeFileSync(filePath, b);
  }
  const form = new FormData();
  form.append('image', fs.createReadStream(filePath), 'tmp.png');
  const response = await fetch('http://localhost:3000/match', { method: 'POST', body: form });
  console.log('status', response.status);
  console.log(await response.text());
}

const { FormData, File } = globalThis;
if (!globalThis.FormData) {
  const { FormData } = require('form-data');
  globalThis.FormData = FormData;
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: repro_upload.py
import os, requests, glob, sys
files = glob.glob(r'.\**\*.png', recursive=True) + glob.glob(r'.\**\*.jpg', recursive=True) + glob.glob(r'.\**\*.jpeg', recursive=True)
files = [f for f in files if os.path.isfile(f)]
print('files', files[:10])
if not files:
    sys.exit(0)
path = files[0]
with open(path, 'rb') as fh:
    data = fh.read()
files_payload = {'image': (os.path.basename(path), data, 'image/jpeg' if path.lower().endswith(('.jpg','.jpeg')) else 'image/png')}
try:
    r = requests.post('http://localhost:3000/match', files=files_payload, timeout=60)
    print('status', r.status_code)
    print(r.text[:4000])
except Exception as e:
    print('ERR', e)
